src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,101 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    public static final int DEFAULT_SIZE = 10;
+    public static final double GROW_RATE = 1.5;
+    private int size;
+    private T[] elementData;
+
+    public ArrayList() {
+        this.elementData = (T[]) new Object[DEFAULT_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        growIfArrayFull();
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        growIfArrayFull();
+        System.arraycopy(elementData,index,elementData,index + 1,size - index);
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
```

Comment:
it's ok, but I would rather do for-each

Reply:
> it's ok, but I would rather do for-each

<img width="625" alt="Screenshot 2023-03-26 at 2 06 55 PM" src="https://user-images.githubusercontent.com/123667530/227795491-df5519d0-ab66-4346-8989-070349001a20.png">

That mistake has been written😱 How to solve it ?

Reply:
please disregard, we used to have for-each loop working on java.util.List class but as here it's our own implementation it's not supported